File: Niveau 3/9 – Efficacité temporelle/2 - Plus long palindrome.py
def main():
    line = input()
    length = len(line)
    actualMaxLength = 1
    for pivot in range(length):
        for i in range(2):
            _length = 1
            end = False
            while not (end):
                end = True
                indiceA = pivot - _length
                indiceB = pivot + _length - (1 if i % 2 == 0 else 0)
                if (indiceA >= 0) and (indiceB < length) and line[indiceA] == line[indiceB]:
                    _length += 1
                    end = False
                elif _length > 1:
                    a = indiceB - indiceA - 1
                    if a > actualMaxLength:
                        actualMaxLength = a
    print(actualMaxLength)


main()

File: Niveau 3/09 – Efficacité temporelle/2 - Plus long palindrome.cpp
#include <iostream>
#include <string>
using namespace std;

int main()
{
  ios::sync_with_stdio(false);
  string s;
  getline(cin, s);
  int length = s.length();
  int actualMaxLength = 1;
  for (int pivot = 0; pivot < length; pivot++)
  {
    for (int i = 0; i < 2; i++)
    {
      int _length = 1;
      bool end = false;
      while (!end)
      {
        end = true;
        int indiceA = pivot - _length;
        int indiceB = pivot + _length - i;
        if ((indiceA >= 0) && (indiceB < length) && s[indiceA] == s[indiceB])
        {
          _length += 1;
          end = false;
        }
        else if (_length > 1)
        {
          int l = indiceB - indiceA - 1;
          if (l > actualMaxLength)
          {
            actualMaxLength = l;
          }
        }
      }
    }
  }
  cout << actualMaxLength;
}